### ISSUE TITLE
Error after upgrading to 1.4.28

### ISSUE THREAD

**Issue by daveg717** [madskristensen/CloseAllTabs#9]
### Installed product versions
- Visual Studio: 2017 Professional 15.8.5 on Win7-64
- This extension: 1.4.28

### Description
After upgrading from 1.4.20 to 1.4.28, clicking on Option -> Environment -> Clean Solution throws an error that the extension could not be loaded, and I see the following in the ActivityLog.  Uninstalling and reverting back to 1.4.20 resolves the issue.

SetSite failed for package [CleanOnClosePackage]Source: &apos;CloseAllTabs&apos; Description: Retrieving the COM class factory for component with CLSID {88AC98C7-B38C-404B-BD86-D2A4F2E89DCA} failed due to the following error: 800401f9 Error in the DLL (Exception from HRESULT: 0x800401F9 (CO_E_ERRORINDLL)).
System.Runtime.InteropServices.COMException (0x800401F9): Retrieving the COM class factory for component with CLSID {88AC98C7-B38C-404B-BD86-D2A4F2E89DCA} failed due to the following error: 800401f9 Error in the DLL (Exception from HRESULT: 0x800401F9 (CO_E_ERRORINDLL)).
   at CloseAllTabs.CloseOpenDocuments..ctor(IServiceProvider serviceProvider, DTE2 dte, Options options) in C:\projects\closealltabs\src\Commands\CloseOpenDocuments.cs:line 22
   at CloseAllTabs.CleanOnClosePackage.<InitializeAsync>d__0.MoveNext() in C:\projects\closealltabs\src\VSPackage.cs:line 27
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.AsyncPackage.<>c__DisplayClass16_0.<<Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask.<JoinAsync>d__78.MoveNext()

### Steps to recreate
1. Upgrade from 1.4.20 to 1.4.28
2. Click on Options -> Environment -> Clean Solution
3. Observe error

### Other extensions installed
Clear MEF Component Cache
Code alignment
Customize Visual Studio Window Title
Disable Solution Explorer's Dynamic Nodes
Dotnet Extensions for Test Explorer
Error Catcher II
Favorite Documents
I Hate #Regions
Match Margin
Microsoft Library Manager
Microsoft Visual Studio Test Platform
Power Commands for Visual Studio
Razor Language Services
Roslyn Expression Evaluators
Roslyn Interactive Components
Roslyn Language Services
Spartez Issue Key Checkins Policy
StopOnFirstBuildError
Structure Visualizer In Map Mode Scrollbar
Trailing Whitespace Visualizer
Visual F# Tools

**Comment by aderrose** [comment]
Hi daveg717,

I'm having the same problem but I can't find where to download version 1.4.20, I've always fetched it direct through VS so don't have any old downloads hanging around.

Have you got a link I could use?

**Comment by Vrlwrox** [comment]
Same thing happens in VS 2015. It also happens when (any) solution is loaded for the first time.

**Comment by modulexcite** [comment]
Hi @aderrose, 
You can always get the source code prior to the [changes](https://github.com/madskristensen/CloseAllTabs/tree/edc69a7c265bf6280238a4ca645e5b137e3d3d4f) and build it to get the previous version. I know it's extra work and not the solution you're looking for but just a thought. 

- Uninstall `CloseAllTabs` from `VS`.

- Download the source code for the previous version from [GitHub](https://github.com/madskristensen/CloseAllTabs/archive/edc69a7c265bf6280238a4ca645e5b137e3d3d4f.zip)

- Extract the `archive`.

- Open the solution in `Visual Studio` and Build.

**Comment by daveg717** [comment]
Hi @aderrose,

I've attached v1.4.20 to this message.  I always download and install extensions manually so I can keep an archive of older versions for situations like this.

[Clean_Solution_v1.4.20.zip](https://github.com/madskristensen/CloseAllTabs/files/2411190/Clean_Solution_v1.4.20.zip)

EDIT: Also, when you originally installed v1.4.20 a copy of the VSIX was created in your user temp directory with the filename VSIX********.vsix.  This file is not automatically deleted, so it should still be there if you haven't cleaned your temp directory.  If you unzip this file, you can open the catalog.json file inside to determine which extension and version it is.

**Comment by aderrose** [comment]
Hi @daveg717,

Thanks for the file, I'll install that shortly, sadly in my attempts to fix the problem I repaired then uninstalled/reinstalled VS and cleaned out various temp/cache folders.

My next step was to reinstall Windows which sounds drastic but I am running the Slow Ring Insider Build at the moment so I wondered if that was causing an issue.

In the future I'll know to be more careful with updates, although until now I've not had a problem.

**Comment by madskristensen** [comment]
I cannot reproduce this issue. I've tried both right after VS opens and also after a solution has loaded.

**Comment by madskristensen** [comment]
Oh, I just saw that 1.4.28 wasn't the latest build. I've made a small change and uploaded version 1.4.30. Can you please try that one and see if it fixes the issue?

**Comment by aderrose** [comment]
I've just had a very quick look and all seems well again, options page opens correctly and no error on solution load and bin and obj folders deleted on close.

I'll have a look at the new features shortly.

**Comment by daveg717** [comment]
1.4.30 resolves the crash issue for me. The deletion of bin and obj folders is working correctly, however my projects in solution explorer are not collapsing after closing and reopening the solution.  Let me know if there is anything I can provide to assist in debugging.

**Comment by modulexcite** [comment]
@madskristensen V1.4.30 works. I just tried it on VS 2017 Pro on Windows 10 x64.
Can you also check if this is an issue with the latest [NodeExecutor](https://github.com/madskristensen/NodeExecutor) as I get an error after updating it to v0.5.7?

**Comment by madskristensen** [comment]
@modulexcite Looks like NodeExecutor has an issue too. I'll upload a new version in a few minutes. Thanks for the heads up

**Comment by MiYanni** [comment]
@daveg717 I'm also seeing no collapsing at all with 1.4.30. I have all the Solution Explorer collapsing options set to `True`, but nothing happens. Additionally, `Close documents` doesn't seem to worth either.
**Edit**: I noticed that I was running into this with net452 projects (VS 2015 style). It seemed to cause issues for my .vs files, and Intellisense in those projects. Not sure if this is related. After I removed the extension and deleted my .vs folder, Intellisense was working as expected.
